functions/index.js

```diff
@@ -255,91 +308,73 @@ const updateUser = (req, res) => {
         });
 };
 
-// User Post
 
-const newPost = (req, res) => {
-    const Post = {
-        username: req.user.username,
-        post: req.body.post,
-        tags: req.body.tags,
-        canLikes: req.body.canLikes,
-        canUnlike: req.body.canUnlike,
+ 
 
+// User Post
+// funciona
+const newPost = (req, res) => {
+    const Postuser = {
+        username: req.body.username ||"",
+        post: req.body.post ||"",
+        tag: req.body.tag ||"",
+        canLikes: req.body.canLikes ||"",
+        canUnlike: req.body.canUnlike ||"",
     };
-    if (Post.post == undefined) {
+    if (Postuser.post == undefined) {
         return res.status(400).json({ error: "Missing require parameter" });
     }
+    console.log("Starting Upload...", Postuser.username);
 
-    console.log("Starting Upload...", newUser.username);
-
-    db.collections("posts")
-        .add(Post)
+    db.doc(`/posts/${Postuser.post}`)
```

Comment:
This is weird, it doesn't make sense to create a new entry in /posts/{post}. If post can be a very long string, I'm not sure it's going to be advantageous to store it like that in the database. 

Instead, do it without the ${Postuser.post} -> db.doc('/posts/')